$OpenBSD$

OpenBSD patch for patch-xftp-web.

$OpenBSD$

Use project-relative `file-embed` paths for the XFTP web assets.
The build should embed the bundled static tree from the extracted sources.

--- apps/xftp-server/XFTPWeb.hs.orig
+++ apps/xftp-server/XFTPWeb.hs
@@ -10,7 +10,7 @@
 import Control.Monad (forM_)
 import qualified Data.ByteString.Char8 as B
 import Data.ByteString (ByteString)
-import Data.FileEmbed (embedDir, embedFile)
+import Data.FileEmbed (embedDir, embedFile, makeRelativeToProject)
 import Data.Maybe (isJust)
 import Data.String (fromString)
 import Web.Embedded (embeddedContent)
@@ -26,13 +26,13 @@
 import System.FilePath ((</>))

 xftpWebContent :: [(FilePath, ByteString)]
-xftpWebContent = $(embedDir "apps/xftp-server/static/xftp-web-bundle/")
+xftpWebContent = $(embedDir =<< makeRelativeToProject "apps/xftp-server/static/xftp-web-bundle/")

 xftpMediaContent :: [(FilePath, ByteString)]
-xftpMediaContent = $(embedDir "apps/xftp-server/static/media/")
+xftpMediaContent = $(embedDir =<< makeRelativeToProject "apps/xftp-server/static/media/")

 xftpFilePageHtml :: ByteString
-xftpFilePageHtml = $(embedFile "apps/xftp-server/static/file.html")
+xftpFilePageHtml = $(embedFile =<< makeRelativeToProject "apps/xftp-server/static/file.html")

 xftpGenerateSite :: XFTPServerConfig s -> Maybe ServerPublicInfo -> Maybe TransportHost -> FilePath -> IO ()
 xftpGenerateSite cfg info onionHost path = do
